### PR TITLE
Using Stackdriver Uptime Checks for Triggering Cloud Functions on a Schedule

### PR DESCRIPTION
**Pull request by sujoyg**
This is a tutorial for scheduling HTTP Cloud Functions using Stackdriver uptime checks.

- [x] Tech writing / styleguide review - @labadav  
- [x] Formatting review for publishing - @jmdobry

### CODE REVIEW

File: tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+
+## Master Cloud Functions
+
+You can address both of these limitations by creating a master Cloud Function for each schedule that triggers all other Cloud Functions for that schedule. The master Cloud Function reads a _trigger list_ file containing all the information required to trigger the associated Cloud Functions. By using a master Cloud Function, you only need to create and manage one uptime check and one master Cloud Function for every schedule.
+
+The master Cloud Function is triggered by Stackdriver uptime checks and maintains a timestamp in a persistent database of when it last triggered the scheduled Cloud Functions. Every time it’s triggered, the master Cloud Function checks the database to see if enough time has elapsed for it to trigger the scheduled functions. If the answer is _yes_, the master Cloud Function updates the timestamp and triggers all the scheduled functions.  If _no_, the functions aren’t triggered.
+
+By using a master Cloud Function in combination with a persistent database, Cloud Functions are executed at most once per interval regardless of the number of uptime check locations in each scheduling interval.
```
Comment (labadav) [inline tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md]: ...Cloud Functions **run** at most once...

File: tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+
+This tutorial implements a trigger list, stored in Cloud Storage, and a master Cloud Function that uses Cloud Datastore to keep track of the timestamp.
+
+# Objectives
+
+Deploy an HTTP Cloud Function triggered on a schedule by:
+* Creating a trigger list in Cloud Storage containing instructions for triggering this function.
```
Comment (labadav) [inline tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md]: ...list in Cloud Storage **that contains** instructions for triggering....

File: tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+    ...
+    status: READY
+    timeout: 60s
+    ...
+    ```
+    A `READY` status indicates that the function has been successfully deployed and is ready to be invoked.
+
+# Creating the trigger list
+
+When the master Cloud Function is executed it invokes the Cloud Functions specified by the trigger list. The trigger list contains a list of URLs corresponding to target HTTP Cloud Functions and the information required to trigger them, such as the HTTP method and any POST data.
```
Comment (labadav) [inline tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md]: The master Cloud Function invokes the Cloud Functions that are specified by the trigger list.

File: tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+                throw 'It is not time yet';
+            }
+        }).catch((e) => {
+            transaction.rollback();
+            throw e;
+        });
+}
+</pre>
+
+If enough time has elapsed, 60 seconds in this example, or if the entity is not found, which is the case the first time this function is executed, `checkTimestamp` does the following:
```
Comment (labadav) [inline tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md]: If enough time has elapsed  (60 seconds in this example), or if the entity is not found

**Comment by sujoyg** [comment]
I approve the commit from @jmdobry

**Comment by jmdobry** [comment]
Published at https://cloud.google.com/community/tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions

Please take a look

**Comment by sujoyg** [comment]
Thank you @jmdobry, looks good.

**Pull request merged**